library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -0,0 +1,125 @@
+# you may find current contact information at www.novell.com
+#
+# ***************************************************************************
+# File: fs_snapshot.rb
+#
+# Authors:
+#	Imobach Gonzalez Sosa <igonzalezsosa@suse.com>
+
+require "yast"
+
+module Yast2
+  class SnapperNotConfigured < StandardError
+    def initialize
+      super "Snapper is not configured."
+    end
+  end
+
+  # Class for managing filesystem snapshots. It's important to note that this
+  # class is intended to be used during installation/update so it uses the
+  # Snapper's CLI because the DBus interface is not available at that time.
+  class FsSnapshot
+    FIND_CONFIG_CMD = "/usr/bin/snapper --no-dbus list-configs | grep \"^root \" >/dev/null"
+    CREATE_CONFIG_CMD = "/usr/bin/snapper --no-dbus create-config -f btrfs /"
+    CREATE_SNAPSHOT_CMD = "/usr/lib/snapper/installation-helper --step 5 --description \"%s\""
+    LIST_SNAPSHOTS_CMD = "LANG=en_US.UTF-8 /usr/bin/snapper --no-dbus list"
+
+    attr_reader :number, :snapshot_type, :previous, :timestamp, :user, :cleanup, :description
+
+    # Determines whether snapper is configured or not
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), FIND_CONFIG_CMD)
+      out["exit"] == 0
+    end
+
+    # Configures snapper
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configure
+      if configured?
+        true
+      else
+        out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), CREATE_CONFIG_CMD)
+        out["exit"] == 0
+      end
+    end
+
+    # Creates a new snapshot
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot,nil] The created snapshot if the operation was
+    #                          successful. Otherwise, it returns nil.
+    def self.create(description)
+      raise SnapperNotConfigured unless configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), CREATE_SNAPSHOT_CMD % description)
+      if out["exit"] == 0
+        find(out["stdout"].to_i) # The CREATE_SNAPSHOT_CMD returns the number of the new snapshot.
+      else
+        nil
+      end
+    end
+
+    # Returns all snapshots
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @return [Array<FsSnapshot>] All snapshots that exist in the system.
+    def self.all
+      raise SnapperNotConfigured unless configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), LIST_SNAPSHOTS_CMD)
+      lines = out["stdout"].split(/\n/)[2..-1] # relevant lines from output.
+      lines.map do |line|
+        data = line.split(/\s*\|\s*/)
+        timestamp = data[3] == "" ? nil : DateTime.parse(data[3])
+        new(data[1].to_i, data[0].to_sym, data[2].to_i, timestamp, data[4],
```

Comment:
Yes, I've renamed to `previous_number` and I've added a `previous` method that retrieve that previous snapshot.